components/formats-gpl/src/loci/formats/in/NDPIReader.java

```diff
@@ -131,14 +131,15 @@ public int fileGroupOption(String id) throws FormatException, IOException {
     }
     else if (getSizeX() <= MAX_SIZE || getSizeY() <= MAX_SIZE) {
       int ifdIndex = getIFDIndex(getCoreIndex(), no);
-      in = new RandomAccessInputStream(currentId);
-      tiffParser = new TiffParser(in);
-      tiffParser.setUse64BitOffsets(true);
-      tiffParser.setYCbCrCorrection(false);
-      byte[] b = tiffParser.getSamples(ifds.get(ifdIndex), buf, x, y, w, h);
-      in.close();
-      tiffParser.getStream().close();
-      return b;
+      try (RandomAccessInputStream s = new RandomAccessInputStream(currentId)) {
+        tiffParser = new TiffParser(s);
+        tiffParser.setUse64BitOffsets(true);
+        tiffParser.setYCbCrCorrection(false);
+        return tiffParser.getSamples(ifds.get(ifdIndex), buf, x, y, w, h);
+      }
+      finally {
+        tiffParser.getStream().close();
```

Comment:
Not for this PR (and I haven't looked into the details) but this idiom makes me think that TiffParser should be made Closeable to enable:

```
  try (TiffParser tiffParser = new TiffParser(currentId)) {
      tiffParser.setUse64BitOffsets(true);
      tiffParser.setYCbCrCorrection(false);
      return tiffParser.getSamples(ifds.get(ifdIndex), buf, x, y, w, h);
  }
```

Reply:
At a quick glance, there are a few places that could benefit from such a logic

```
sbesson@ls30630:bioformats (develop) $ git grep getStream.*close
components/formats-gpl/src/loci/formats/in/APLReader.java:            p.getStream().close();
components/formats-gpl/src/loci/formats/in/CellSensReader.java:        parser.getStream().close();
components/formats-gpl/src/loci/formats/in/NDPIReader.java:      tiffParser.getStream().close();
components/formats-gpl/src/loci/formats/in/NDPIReader.java:        tiffParser.getStream().close();
components/formats-gpl/utils/WritePrecompressedPlanes.java:    saver.getStream().close();
```

Extending `TiffParser` (and `TiffSaver`?) to be `Closeable` is certainly something worth discussing in the context of the next minor release especially given the emphasis on the file leak. I do not know of obvious caveats to doing so.